docs/deliverables-portal.md

```diff
@@ -0,0 +1,134 @@
+
+The Deliverables portal will be a page in Sampuru where internal users are able to input/update deliverable information pertinent to a case. The proposed design for the portal is an editable table with the following columns:
+
+| Project*                                                         | Case ID*                                                   | Location* | Notes     | Expiry Date |
+|------------------------------------------------------------------|------------------------------------------------------------|-----------|-----------|-------------|
+| Single select dropdown, selection populates the Case ID dropdown | Multi select dropdown, empty until the project is selected | Free text | Free text | Date picker |
+*indicates a required field
+
+The table will be populated with data from the `/deliverables` endpoint. It's important to note that each row will be editable, meaning for each existing row of deliverables information, the user can modify the cells. With a 'Submit' or 'Save Changes' button, the user should be able to then POST any changes to the `/update_deliverable` endpoint. The rationale behind this design decision is that it allows users to post all of the deliverable information that they want at once, rather than resubmitting for every new piece of information.
+
+This portal was developed using the branch `GR-1376_front-end-deliverables-portal`.
+ 
+
+## Getting Started
+### Sampuru Environment
+Below is a breakdown of some of the important files in the Sampuru code base that concern the deliverables portal, and the role that they each play:
+
+- **deliverables.ts**: TypeScript file where the code relevant to the elements on the deliverables page lives. This page is where the table itself is instantiated and populated with data from the /deliverables endpoint. A brief rundown of the code within this file is as follows:
+
+1. Function to construct the table rows. This function takes `deliverablesObject` as a parameter, which is an object of `DeliverableObject[]` (a data-transfer object). This function essentially where the table rows are generated, and where the deliverables data is pushed into the rows. 
+````
+function constructTableRows(deliverableObjects: DeliverableObject[]): ComplexElement<HTMLTableRowElement>[]{
+const tableRows: ComplexElement<HTMLTableRowElement>[] = [];
+
+
+       Array.from(deliverableObjects).forEach((deliverable) => {
+            deliverable.project_cases.forEach((project_case) => {
+                project_case.cases.forEach((donor_case) => {
+                    deliverable.deliverables.forEach((deliverables) => {
+                        tableRows.push(tableRow(null,
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"}))
+                      })
```

Comment:
I don't think the actual code will have so many of these 'testDummyData' lines in it? Use pseudocode if you have to, but i'm actually more confused by these repeated lines